virtual-server/examples/nodejs/virtual-server-client-examples/client.js

```diff
@@ -0,0 +1,136 @@
+      }
+      return this.kubeclient.apis['virtualservers.coreweave.com'].v1alpha1.namespaces(namespace).virtualservers(name).get()
+    },
+    list: ({namespace}) => {
+      if(!namespace) {
+        return Promise.reject(new Error("Namespace is required"))
+      }
+      return this.kubeclient.apis['virtualservers.coreweave.com'].v1alpha1.namespaces(namespace).virtualservers().get()
+    },
+    // Create a new VirtualServer
+    // manifest is a yaml string
+    create: (manifest) => {
+      const manifestObj = YAML.parse(manifest)
+      if(!manifestObj.metadata.namespace) {
+        return Promise.reject(new Error("VirtualServer metadata.namespace is required"))
+      }
+      return this.kubeclient.apis['virtualservers.coreweave.com'].v1alpha1.namespaces(manifestObj.metadata.namespace).virtualservers.post({body: manifestObj})
+    },
+    // Delete a VirtualServer
+    delete: ({namespace, name}) => {
+      if(!namespace || !name) {
+        return Promise.reject(new Error("Virtual Server namespace and name are required"))
+      }
+      return this.kubeclient.apis['virtualservers.coreweave.com'].v1alpha1.namespaces(namespace).virtualservers(name).delete()
+    },
+    // Update a new VirtualServer
+    // manifest is a yaml string
+    update: async (manifest) => {
+      const manifestObj = YAML.parse(manifest)
+      if(!manifestObj.metadata.namespace || !manifestObj.metadata.name) {
+        return Promise.reject(new Error("VirtualServer metadata.namespace and metadata.name is required"))
+      }
+      if(!manifestObj.metadata.resourceVersion) {
+        await this.virtualServer.get({namespace: manifestObj.metadata.namespace, name: manifestObj.metadata.name}).then(o => manifestObj.metadata.resourceVersion = o.body.metadata.resourceVersion)
+      }
+      return this.kubeclient.apis['virtualservers.coreweave.com'].v1alpha1.namespaces(manifestObj.metadata.namespace).virtualservers(manifestObj.metadata.name).put({body: manifestObj})
+    },
+    // Ready will resolve when the VirtualServer is ready (determined by the status field of the Ready Condition)
+    ready: async({namespace, name}) => {
+      if(!namespace || !name) {
+        return Promise.reject(new Error("Virtual Server namespace and name are required"))
+      }
+      const stream = await this.kubeclient.apis['virtualservers.coreweave.com'].v1alpha1.watch.namespaces(namespace).virtualservers(name).getObjectStream()
+      return new Promise((resolve, reject) => {
+        stream.on('data', async e => {
+          if(e.type === 'DELETED') {
+            stream.end()
+            return reject(new Error(`VirtualServer ${namespace}/${name} was deleted`))
+          }
+          else {
+            if(!!e.object.status) {
+              const readyCondition = e.object.status.conditions.filter(c => c.type === 'Ready')[0] || {}
+              if(readyCondition.status === 'True') {
```

Comment:
Despite proper vs status:
```
NAME                STATUS               REASON               STARTED
my-virtual-server   VirtualServerReady   VirtualServerReady   True
```
the condition does not give us working VMI:
```
  Conditions:
    Last Transition Time:  2021-03-05T10:16:02Z
    Message:               VirtualServerReady
    Reason:                VirtualServerReady
    Status:                True
    Type:                  Ready
    Last Transition Time:  2021-03-05T10:16:02Z
    Message:               ServicesReady
    Reason:                ServicesReady
    Status:                True
    Type:                  ServicesReady
    Last Transition Time:  2021-03-05T10:16:02Z
    Message:               0/1384 nodes are available: 1162 Insufficient nvidia.com/gpu ...
    Reason:                Pending
    Status:                False
    Type:                  VirtualMachineReady
    Last Transition Time:  2021-03-05T10:16:20Z
    Message:               VirtualServerStarted
    Reason:                VirtualServerStarted
    Status:                True
    Type:                  VirtualServerStarted
  Network:
    Tcp IP:  207.53.234.148

```
Is this OK?

Reply:
> 0/1384 nodes are available: 1162 Insufficient nvidia.com/gpu ...
>     Reason:                Pending
>     Status:                False
>     Type:                  VirtualMachineReady

This has been fixed in recent commit